### PR TITLE
irmin-pack: Fix inode's find starting depth

### PR DESCRIPTION
**Pull request by Ngoguey42**

### CODE REVIEW

File: src/irmin-pack/inode.ml [hunk 1]
```diff
@@ -922,20 +922,18 @@ struct
...
             let x = t.entries.(i) in
-            match x with
-            | None -> None
-            | Some i -> aux ~depth:(depth + 1) (target_of_ptr i).v)
+            match x with None -> None | Some i -> aux (target_of_ptr i).v)
       in
-      aux ~depth t.v
+      aux t.v
 
-    let find ?(cache = true) layout t s = find_value ~cache ~depth:0 layout t s
```
Comment (Ngoguey42) [inline src/irmin-pack/inode.ml:938]: Depth might not be 0 if we're in the situation where `Val.t` is not a root

File: src/irmin-pack/inode.ml [hunk 2]
```diff
@@ -1029,7 +1027,7 @@ struct
 
     let remove layout t s =
       (* XXX: [find_value ~depth:42] should break the unit tests. It doesn't. *)
```
Comment (CraigFe) [inline src/irmin-pack/inode.ml:1029]: This comment is now meaningless.

**Review by CraigFe — APPROVED** [review]
LGTM, but I'm no expert in Inode's depth-tracking code.

File: src/irmin-pack/inode.ml (hunk 2 above)
Comment (Ngoguey42) [reply src/irmin-pack/inode.ml]: Thanks

**Comment by Ngoguey42** [comment]
Let's merge as soon as CI is happy

**Comment by CraigFe** [comment]
Pushed a formatting fix.

**Comment by Ngoguey42** [comment]
Thanks

**Pull request merged**